nvflare/fuel/utils/config_factory.py

```diff
@@ -47,23 +47,31 @@ def search_config_format(
             search_dirs: search directory. If none, the parent directory of init_file_path will be used as search dir
 
         Returns:
-            Tuple of None,None or ConfigFormat and real configuration path
+            Tuple of None,None or ConfigFormat and real configuration file path
 
         """
         logger = ConfigFactory.logger
-        if not search_dirs:
+        if not search_dirs:  # empty or None
             parent_dir = pathlib.Path(init_file_path).parent
             search_dirs = [str(parent_dir)]
+
         # we ignore the original extension
         file_basename = os.path.splitext(pathlib.Path(init_file_path).name)[0]
         ext2fmt_map = ConfigFormat.config_ext_formats()
         for search_dir in search_dirs:
-            logger.debug(f"search file:{file_basename} basename, search dirs = {search_dirs}")
+            logger.debug(f"search file basename:'{file_basename}', search dirs = {search_dirs}")
             for ext in ext2fmt_map:
                 fmt = ext2fmt_map[ext]
-                file = os.path.join(search_dir, file_basename, ext)
-                if os.path.isfile:
-                    return fmt, file
+                file_name = f"{file_basename}{ext}"
+                config_file = os.path.join(search_dir, file_name)
+                if os.path.isfile(config_file):
+                    return fmt, config_file
+                else:
+                    # check child directories:
+                    for root, dirs, files in os.walk(search_dir):
+                        if file_name in files:
+                            config_file = os.path.join(root, file_name)
+                            return fmt, config_file
```

Comment:
This can be simplified as:

                parent_dir = str(pathlib.Path(search_dir).parent)
                for root, dirs, files in os.walk(parent_dir):
                    if file_name in files:
                        config_file = os.path.join(root, file_name)
                        return fmt, config_file

Reply:
By changing to "os.walk(parent_dir)", no need to check the "if os.path.isfile(config_file)" particularly any more. That will be covered by the "os.walk(parent_dir)".